pkg/controller/jaeger/account.go

```diff
@@ -3,15 +3,16 @@ package jaeger
 import (
 	"context"
 
+	log "github.com/sirupsen/logrus"
 	corev1 "k8s.io/api/core/v1"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 
-	"github.com/jaegertracing/jaeger-operator/pkg/apis/jaegertracing/v1"
+	v1 "github.com/jaegertracing/jaeger-operator/pkg/apis/jaegertracing/v1"
 	"github.com/jaegertracing/jaeger-operator/pkg/inventory"
 )
 
 func (r *ReconcileJaeger) applyAccounts(jaeger v1.Jaeger, desired []corev1.ServiceAccount) error {
-	opts := client.MatchingLabels(map[string]string{
+	opts := client.InNamespace(jaeger.Namespace).MatchingLabels(map[string]string{
```

Comment:
This (and similar changes in the other `controller` functions) is the actual fix for the reported problem. All other changes are either to make the test more accurate, or to properly log what's going on.